apis/v1alpha1/gateway_types.go

```diff
@@ -295,15 +295,11 @@ const (
 // should raise a "Detached" condition for the affected Listener with
 // a reason of "UnsupportedProtocol".
 //
-// Valid ProtocolType values are:
-//
-// * "HTTP"
-// * "HTTPS"
-// * "TLS"
-// * "TCP"
-// * "UDP"
-//
-// +kubebuilder:validation:Enum=HTTP;HTTPS;TLS;TCP;UDP
+// Implementations can define their own protocols if one is not already defined.
+// Un-prefixed names are reserved for protocols defined in this package.
+// Non-standard protocols should use prefixed names such as
+// mycompany.com/my-custom-protocol.
+// Any protocol defined by implementations will fall under custom conformance.
```

Comment:
I think we probably still want to retain a list of core protocol types that we support. Maybe just replace `Valid ProtocolType values...` with `Core ProtocolType values...`. Then you can add the paragraph about custom protocols below that.

Reply:
Reworded this a little bit. PTAL.